pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -222,10 +224,16 @@ func (r *ReconcileComplianceScan) validate(instance *compv1alpha1.ComplianceScan
 		return false, nil
 	}
 
-	// Set default scan type if missing
-	if instance.Spec.ScanType == "" {
+	if instance.Spec.ScanType == "" || instance.Spec.ScannerType == "" {
 		instanceCopy := instance.DeepCopy()
-		instanceCopy.Spec.ScanType = compv1alpha1.ScanTypeNode
+		// Set default scan type if missing
+		if instance.Spec.ScanType == "" {
+			instanceCopy.Spec.ScanType = compv1alpha1.ScanTypeNode
+		}
+		// Set default scanner type if missing
+		if instance.Spec.ScannerType == "" {
```

Comment:
nit: Instead of performing the same check twice, we could pull this out of the conditional on line 227.

```go

if instance.Spec.ScanType == "" {
  // Set node scan type
}
if instance.Spec.ScannerType == "" {
  // Set SCAP scanner type
}

```